test/Tests/src/Data/List_Spec.enso

```diff
@@ -38,6 +38,44 @@ spec = Test.group "List" <|
         l.contains 4 . should_be_false
         l.contains 3 . should_be_true
         empty.contains 10 . should_be_false
+    Test.specify "should allow finding an element in the list with `.find`" <|
+        l.find (==2) . should_equal 2
+        l.find (==3) . should_equal 3
+        l.find (==4) . should_fail_with Not_Found
+        l.find (==2) start=1 . should_equal 2
+        l.find (==2) start=2 . should_fail_with Not_Found
+        l.find (==2) start=-1 . should_fail_with Not_Found
+        l.find (==2) start=-2 . should_equal 2
+        empty.find (==1) . should_fail_with Not_Found
+        empty.find (==1) if_missing=Nothing . should_equal Nothing
```

Comment:
What if I do `start=100`? We should have a test for `find` that checks out of bounds offsets, much like we have for `index_of`.